keps/sig-node/2008-forensic-container-checkpointing/README.md

```diff
@@ -292,14 +329,94 @@ Checkpointing containers will be possible again.
 
 ###### Are there any tests for feature enablement/disablement?
 
-Currently no.
+Currently the test will automatically be skipped if the feature is not enabled.
+
+### Rollout, Upgrade and Rollback Planning
+
+Does not apply as the feature is an additional API endpoint with no
+dependencies on other functionality. If it is not enabled via the feature
+gate it will return `404 page not found`. If it is not enabled in the
+underlying container engine a `500` will be returned with an error
```

Comment:
What guarantees that all container engines do this?  Is the "no action" behavior from containerd (and all other container engines) a 500?  If there isn't uniform support after two years, are we adding a kubelet API for one implementation?

Reply:
> I want to know if I take a kubelet today, put it against an old containerd that doesn't have this feature, how does it behave. is it the container engine returning a 500 or the kubelet detecting a failrue and returning a 500.

It behaves like every other CRI call that is not implemented by the container engine (500):
```
(rpc error: code = Unimplemented desc = unknown method CheckpointContainer for service runtime.v1.RuntimeService)
```

> In a kep with no implementation details, it's nearly impossible to tell.

Up until now it was not clear that a KEP is about implementation details. Until now it was always about high level design from my understanding. I will add implementation details as mentioned above. At this point it is just passing through the kubelet API endpoint call to the CRI.